packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/Statements/PragmaWarningRestoreStatement.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.TypeSpec.Generator.Expressions;
+using Microsoft.TypeSpec.Generator.Snippets;
+
+namespace Microsoft.TypeSpec.Generator.Statements
+{
+    public class PragmaWarningRestoreStatement : MethodBodyStatement
+    {
+        public ValueExpression Code { get; }
+        public string Justification { get; }
+
+        public PragmaWarningRestoreStatement(ValueExpression code, string justification)
```

Comment:
Why did we need the separate PragmaWarningRestore/Disable statements as opposed to using SuppressionStatement?

Reply:
Nevermind - I see that this is needed for the constructor suppression.